bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -827,4 +826,158 @@ private void handlePushActivity(@Nullable String payload) {
+                child.setDeviceAndUpdateThingState(this, findSmartDeviceHomeJson(id));
+            }
+        }
+        if (smartHomeDevices != null) {
+            Map<String, SmartHomeDevice> newJsonSerialDeviceMapping = new HashMap<>();
+            for (Object smartDevice : smartHomeDevices) {
+                if (smartDevice instanceof SmartHomeDevice) {
+                    SmartHomeDevice shd = (SmartHomeDevice) smartDevice;
+                    String entityId = shd.entityId;
+                    if (entityId != null) {
+                        newJsonSerialDeviceMapping.put(entityId, shd);
+                    }
+                }
+            }
+        }
+        if (smartHomeDevices != null) {
+            return smartHomeDevices;
+        }
+
+        return new ArrayList<SmartHomeBaseDevice>();
+    }
+
+    public void forceDelayedSmartHomeStateUpdate() {
+        synchronized (synchronizeSmartHomeJobScheduler) {
+            ScheduledFuture<?> refreshSmartHomeAfterCommandJob = this.refreshSmartHomeAfterCommandJob;
+            if (refreshSmartHomeAfterCommandJob != null) {
+                refreshSmartHomeAfterCommandJob.cancel(false);
+            }
+            this.refreshSmartHomeAfterCommandJob = scheduler.schedule(this::updateSmartHomeStateJob, 500,
+                    TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void updateSmartHomeStateJob() {
+        synchronized (synchronizeSmartHomeJobScheduler) {
+            this.refreshSmartHomeAfterCommandJob = null;
+        }
+        updateSmartHomeState();
+    }
+
+    synchronized private void updateSmartHomeState() {
+        if (refreshSmartHomeAfterCommandJob != null) {
+            logger.debug("updateSmartHomeState suspended");
+            return;
+        }
+        try {
+            logger.debug("updateSmartHomeState started");
+            Connection connection = this.connection;
+            if (connection == null || !connection.getIsLoggedIn()) {
+                return;
+            }
+            List<SmartHomeBaseDevice> allDevices = this.getLastKnownSmartHomeDevice();
+            Set<String> applianceIds = new HashSet<>();
+            synchronized (this.smartHomeDeviceHandlers) {
+                if (this.smartHomeDeviceHandlers.size() == 0) {
+                    return;
+                }
+                for (SmartHomeDeviceHandler device : smartHomeDeviceHandlers) {
+                    String id = device.findId();
+                    if (id != null) {
+                        SmartHomeBaseDevice baseDevice = jsonIdSmartHomeDeviceMapping.get(id);
+                        for (SmartHomeDevice shd : SmartHomeDeviceHandler.GetSupportedSmartHomeDevices(baseDevice,
+                                allDevices)) {
+                            applianceIds.add(shd.applianceId);
+                        }
+                    }
+                }
+            }
+
+            Map<String, JsonArray> applianceIdToCapabilityStates = connection
```

Comment:
A single call to the amazon server returns now all states of all defined devices